### PR TITLE
Catch2: support v3

### PR DESCRIPTION
**Pull request by jnewb1**
original work from: https://github.com/pytest-dev/pytest-cpp/pull/107

### CODE REVIEW

File: tests/test_pytest_cpp.py [hunk 1]
```diff
@@ -596,26 +601,32 @@ def test_cpp_verbose(testdir, exes):
...
+    for suffix in ["", "_v3"]:
+        facade = Catch2Facade()
+        failures, _ = facade.run_test(
+            exes.get(f"catch2_failure{suffix}"), "Factorials are computed"
+        )
+        assert len(failures) == 1
+
+        fail1 = failures[0]
+        colors = ("red", "bold")
+        assert_catch2_failure(fail1.get_lines()[0], "Expected: ", colors)
```
Comment (jnewb1) [inline tests/test_pytest_cpp.py:619]: seems like the whitespace changed between v2 and v3

File: src/pytest_cpp/catch2.py [hunk 2]
```diff
@@ -20,11 +21,11 @@ class Catch2Facade(AbstractFacade):
     """
 
     @classmethod
-    def is_test_suite(
+    def get_catch_version(
         cls,
         executable: str,
         harness_collect: Sequence[str] = (),
-    ) -> bool:
+    ) -> Optional[str]:
```
Comment (nicoddemus) [inline src/pytest_cpp/catch2.py:28]: If we want to type this correctly, we can either use an enum, or `Literal`:

```suggestion
    ) -> Optional[Literal["v2", "v3"]]:
```

File: src/pytest_cpp/catch2.py [hunk 3]
```diff
@@ -123,11 +148,16 @@ def run_test(
         return [failure], output
 
     def _parse_xml(
-        self, xml_filename: str
+        self, xml_filename: str, catch_version: str
```
Comment (nicoddemus) [inline src/pytest_cpp/catch2.py:151]:
```suggestion
        self, xml_filename: str, catch_version: Literal["v2", "v3"]
```

**Review by nicoddemus — APPROVED** [review]
This looks great @jnewb1, thanks! Can you please also add a CHANGELOG entry?

**Comment by nicoddemus** [comment]
@jnewb1 this is your 2nd high-quality PR to this repository, would you like to be added as a maintainer? It would not entail much more responsibility, but it would give you permission to also review other PRs and handle issues (this is a low volume repository).

**Comment by jnewb1** [comment]
> @jnewb1 this is your 2nd high-quality PR to this repository, would you like to be added as a maintainer? It would not entail much more responsibility, but it would give you permission to also review other PRs and handle issues (this is a low volume repository).

sure, you can add me as a maintainer!

**Review by nicoddemus — APPROVED** [review]
Thanks!

**Comment by nicoddemus** [comment]
> sure, you can add me as a maintainer!

Awesome! Sent you an invite. 👍

**Pull request merged**